pkg/sysched/sysched.go

```diff
@@ -0,0 +1,492 @@
+// system call set is return for the pod
+func (sc *SySched) getSyscalls(pod *v1.Pod) ([]string, error) {
+	var r []string
+
+	// read the seccomp profile from the security context of a pod
+	podSC := pod.Spec.SecurityContext
+	if podSC != nil && podSC.SeccompProfile != nil && podSC.SeccompProfile.Type == "Localhost" {
+		profilePath := *podSC.SeccompProfile.LocalhostProfile
+		ns, crdName := getCRDandNamespace(profilePath)
+		//klog.Info("namespace and profile name from pod sc = ", ns, " ", crdName)
+
+		syscalls, _ := sc.readSPOProfileCRD(crdName, ns)
+
+		if len(syscalls) > 0 {
+			r = UnionList(r, syscalls)
+		}
+	}
+
+	// read the seccomp profile from container security context and merge them
+	for _, container := range pod.Spec.Containers {
+		conSC := container.SecurityContext
+		if conSC != nil && conSC.SeccompProfile != nil && conSC.SeccompProfile.Type == "Localhost" {
+			profilePath := *conSC.SeccompProfile.LocalhostProfile
+			ns, crdName := getCRDandNamespace(profilePath)
+			//klog.Info("namespace and profile name from pod sc = ", ns, " ", crdName)
+
+			syscalls, _ := sc.readSPOProfileCRD(crdName, ns)
+
+			if len(syscalls) > 0 {
+				r = UnionList(r, syscalls)
+			}
+		}
+	}
+
+	// SPO seccomp profiles are sometimes automatically annotated to a pod
+	if pod.ObjectMeta.Annotations != nil {
+		// there could be multiple SPO seccomp profile annotations for a pod
+		// merge all profiles to obtain the syscal set for a pod
+		for k, v := range pod.ObjectMeta.Annotations {
+			// looks for annotation related to the seccomp
+			if strings.Contains(k, "seccomp.security.alpha.kubernetes.io") {
+				ns, crdName := getCRDandNamespace(v)
+				//klog.Debug("namespace and name = ", ns, crdName)
+
+				syscalls, _ := sc.readSPOProfileCRD(crdName, ns)
+
+				// look for next annotation if the profile is empty
+				if len(syscalls) == 0 {
+					continue
+				}
+
+				// merge if there are multiple profiles
+				r = UnionList(r, syscalls)
+			}
+		}
+	}
+
+	// if a pod does not have a seccomp profile specified, return unconfined syscall set
+	if len(r) == 0 {
+		r, _ = sc.getUnconfinedSyscalls()
+	}
+
+	//klog.Info("number of system calls for pod = ", len(r))
+	return r, nil
+}
+
+// Returns all available system calls by reading the full SPO seccomp profile CRD
+func (sc *SySched) getUnconfinedSyscalls() ([]string, error) {
+	var r []string
+
+	// extract a seccomp crd using namespace and crd name
+	r, _ = sc.readSPOProfileCRD(sc.FullSyscallProfile, sc.SyscallCRDNamespace)
+
+	// klog.Info("unconfined system calls for pod = ", r)
+	return r, nil
+}
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (sc *SySched) Name() string {
+	return Name
+}
+
+func (sc *SySched) calcScore(syscalls []string) int {
+	tot_crit := 0
+
+	// TODO: weight for critical/cve syscalls
+	W := 1
+	score := len(syscalls) - tot_crit
+	score = score + W*tot_crit
+	klog.Info(score, " ", tot_crit)
```

Comment:
I think this function will always return same score. `tot_crit` is always 0

Reply:
Currently, our plugin does not support syscall weights. We are working on it and update the weighting in our next version. We leave that portion of code intentionally. Also, we have put a note there regarding it.